src/batterywatcher.cpp

```diff
@@ -64,7 +64,8 @@ BatteryWatcher::BatteryWatcher(QObject *parent) : Watcher(parent)
 
     mBatteryInfoDialog = new BatteryInfoDialog(mBatteries);
 
-    connect(&mSettings, SIGNAL(settingsChanged()), this, SLOT(settingsChanged()));
+    connect(&mSettings, &PowerManagementSettings::settingsChanged, this, &BatteryWatcher::settingsChanged);
+    // connect(LXQt::Settings::globalSettings(), &LXQt::GlobalSettings::iconThemeChanged, this, &BatteryWatcher::settingsChanged);
     connect(LXQt::Settings::globalSettings(), SIGNAL(iconThemeChanged()), this, SLOT(settingsChanged()));
```

Comment:
I think the problem is in liblxqt and the string-based syntax may have had a hidden issue too.

@luis-pereira, would you please take a look at this?

Reply:
IMO this is because of the internal linkage and that symbol is not exported. This one can't  be migrated to new syntax without a change in liblxqt...

Reply:
> This one can't be migrated to new syntax without a change in liblxqt

Right.

I didn't find any indication in Qt's doc that an old string-based syntax may not be able to be converted to the new one. That's why I think that our old-style connection may also have a problem — not a good argument but just a feeling.

Reply:
PR for liblxqt done -> lxqt/liblxqt#278